wordpress-mobile/WordPress-Android
# Blaze: Add menu item entry point

Parent #17910 

This PR adds the Blaze Menu item entry point

- Checks for eligibilty using `BlazeFeatureUtils.shouldShowBlazeEntryPoint`
- Shows the menu item on the home tab
- Tracks the event tapped

Light | Dark
-- | --
<img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/222542267-f430e5a5-8923-4b11-a5e1-76dedd26ca61.png"> | <img width="250" height="500" alt="Alt desc" src="https://user-images.githubusercontent.com/506707/222542256-f53bd339-d975-489f-bac2-258b1ebf8f69.png">

**To test:**
- Launch the app
- Login with a wp.com account
- Select a site that is blaze eligible
- Navigate to the Home tab
- ✅ Verify the Blaze menu item is visible
- Tap the Blaze menu item
- ✅ Verify you see the Blaze Overlay
- ✅ Verify the logs contain 🔵 Tracked: blaze_feature_tapped, Properties: {"source":"menu_item"}

## Regression Notes
1. Potential unintended areas of impact
The blaze menu item shows when it shouldn't

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Updated existing tests; MySiteViewModelTest

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
